cmds/tui.c

```diff
@@ -988,6 +988,22 @@ static void win_header_graph(struct tui_window *win,
 	memcpy(graph->disp_mask, graph->top_mask, graph->mask_size);
 }
 
+static void win_footer(struct tui_window *win, char *msg)
+{
+	char footer[COLS + 1];
+	char buf[COLS + 1];
```

Comment:
Do we really need two buffers for this?

Reply:
I was trying to just use the same argument (`char *msg`) instead of creating new buffer, 

But in next commit, for the debug message
prior to `snprintf`, I call memset to remove NULL character.
```
	if (!is_graph && tui_debug) {
		/* for removing null character */
		memset(buf, KEY_SPACE, COLS);
		snprintf(buf, COLS, "uftrace debug: top: %d, curr: %d",
			 win->top_index, win->curr_index);
	}
```

Well, using the same string buffer with pointer is okay, 
but I thought that it seems weird to memset the pointer buffer without checking the size.

From the point of `win_footer`'s view, it doesn't know how size of the `char *msg` buffer has, 
(well, we know it has [COLS+1] byte since we've defined it)

So copying the `*msg` to a new buffer seems more secure to me.

Reply:
Why not doing something like below?

```c
if (debug)
    snprintf(buf, len, "uftrace debug: ...", ...);
else
    strncpy(buf, len, "%s", msg);
```